tests/aws/services/route53resolver/test_route53resolver.py

```diff
@@ -721,3 +722,221 @@ def test_list_firewall_domain_lists(self, cleanups, snapshot, aws_client):
+        )
+
+        with pytest.raises(
+            aws_client.route53resolver.exceptions.ResourceNotFoundException
+        ) as resource_not_found:
+            aws_client.route53resolver.list_firewall_rules(
+                FirewallRuleGroupId=missing_firewall_rule_group_id
+            )
+
+        error_msg = resource_not_found.value.response["Error"]["Message"]
+        match = re.search("Trace Id: [\"'](.+)[\"']", error_msg)
+        if match:
+            trace_id = match.groups()[0]
+            snapshot.add_transformer(snapshot.transform.regex(trace_id, "<trace-id>"))
+
+        snapshot.match("missing-firewall-rule-group-id", resource_not_found.value.response)
+
+    @markers.aws.validated
+    def test_list_firewall_rules_for_empty_rule_group(self, cleanups, snapshot, aws_client):
+        snapshot.add_transformer(snapshot.transform.route53resolver_api())
+        snapshot.add_transformer(snapshot.transform.key_value("Name"))
+
+        rule_group_response = aws_client.route53resolver.create_firewall_rule_group(
+            Name=f"empty-{short_uid()}"
+        )
+        snapshot.match("create-firewall-rule-group", rule_group_response)
+
+        response = aws_client.route53resolver.list_firewall_rules(
+            FirewallRuleGroupId=rule_group_response["FirewallRuleGroup"]["Id"]
+        )
+        assert len(response["FirewallRules"]) == 0, "Expecting 0 firewall rules"
+        snapshot.match("empty-firewall-rule-group", response)
+
+        cleanups.append(
+            lambda: aws_client.route53resolver.delete_firewall_rule_group(
+                FirewallRuleGroupId=rule_group_response["FirewallRuleGroup"]["Id"]
+            )
+        )
+
+    @markers.aws.validated
+    @markers.snapshot.skip_snapshot_verify(paths=["$..ManagedOwnerName"])
+    @markers.snapshot.skip_snapshot_verify(paths=["$..FirewallDomainRedirectionAction"])
+    def test_list_firewall_rules(self, cleanups, snapshot, aws_client):
+        """Test listing firewall rules.
+
+        Setup of this testcase consists in creating:
+         - 1 firewall-rule-group
+         - 4 firewall-domain-lists
+         - 4 firewall-rules
+
+         We test listing:
+         - all rules in the rule-group
+         - rules filtered by priority
+         - rules filtered by action
+         - rules filtered by priority and action
+        """
+        firewall_rule_group_name = f"fw-rule-group-{short_uid()}"
+
+        snapshot.add_transformer(snapshot.transform.route53resolver_api())
+        snapshot.add_transformer(snapshot.transform.key_value("Name"))
+        snapshot.add_transformer(
+            snapshot.transform.key_value("FirewallDomainListId", "fw-domain-list")
+        )
+
+        rule_group_response = aws_client.route53resolver.create_firewall_rule_group(
+            Name=firewall_rule_group_name
+        )
+        snapshot.match("create-firewall-rule-group", rule_group_response)
```

Comment:
We don’t need to duplicate the snapshot of the firewall group creation response here, as it’s already covered in `test_list_firewall_rules_for_empty_rule_group`.

Reply:
Removed.